haystack/components/routers/zero_shot_text_router.py

```diff
@@ -0,0 +1,177 @@
+from typing import Any, Dict, List, Optional
+
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.lazy_imports import LazyImport
+from haystack.utils import ComponentDevice, Secret, deserialize_secrets_inplace
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install transformers[torch,sentencepiece]'") as torch_and_transformers_import:
+    from transformers import pipeline
+
+    from haystack.utils.hf import (  # pylint: disable=ungrouped-imports
+        deserialize_hf_model_kwargs,
+        resolve_hf_pipeline_kwargs,
+        serialize_hf_model_kwargs,
+    )
+
+
+@component
+class TransformersZeroShotTextRouter:
+    """
+    Routes a text input onto different output connections depending on which label it has been categorized into.
+    This is useful for routing queries to different models in a pipeline depending on their categorization.
+    The set of labels to be used for categorization can be specified.
+
+    Example usage in a retrieval pipeline that passes question-like queries to an embedding retriever and keyword-like
+    queries to a BM25 retriever:
+
```

Comment:
I'm still a bit confused about this example 😓 where is the bm25 retriever here? I can see two sentence transformer embedders. Perhaps just a bit more clarification around this example would be super beneficial for users 🙏

Reply:
Ahh okay, that's a good point, I was being a bit lazy in adding the retrievers, since I didn't want to set up a doc store with docs and then also specify the components for each of the retrievers 😅 But I can go ahead and do that.